### PR TITLE
275839 - Owner update endpoint

### PR DESCRIPTION
**Pull request by sfitz42**

### CODE REVIEW

File: app/dto/person.js [hunk 1]
```diff
@@ -1,3 +1,20 @@
+const addContacts = (contacts) => {
+  const emails = contacts.filter(entry => entry.contact.contact_type.contact_type === 'Email')
+    .sort((a, b) => b.id - a.id)
+
+  const primaryTelephones = contacts.filter(entry => entry.contact.contact_type.contact_type === 'Phone')
+    .sort((a, b) => b.id - a.id)
+
+  const secondaryTelephones = contacts.filter(entry => entry.contact.contact_type.contact_type === 'SecondaryPhone')
```
Comment (jbarnsley10) [inline app/dto/person.js:8]: I assumed we could unlimited 'Phone' numbers, just displayed in 'id' order. This limits us to 2 phones - are you happy with that? I think the UI can show multiple (more than 2) phones if they exist

File: app/repos/people.js [hunk 2]
```diff
@@ -109,7 +185,7 @@ const getPersonAndDogsByReference = async (reference, transaction) => {
         {
           model: sequelize.models.person_contact,
           as: 'person_contacts',
-          separate: true, // workaround to prevent 'contact_type_id' being truncated to 'contact_type_i'
```
Comment (jbarnsley10) [inline app/repos/people.js:112]: I would argue this is quite an important comment to keep in, explaining why the 'separate: true' is used. Did you mean to remove?

File: app/dto/person.js (hunk 1 above)
Comment (sfitz42) [reply app/dto/person.js:8]: We can store unlimited phone numbers. It's just a bit tricky as the user can only ever insert two phone numbers at the moment on the update page.

We also need the types to separated for that screen.

File: app/repos/people.js (hunk 2 above)
Comment (sfitz42) [reply app/repos/people.js:112]: Comments like that are better off to have in documentation. Can be a bit messy keeping inline comments like that in in my opinion.

File: app/schema/person/update.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const Joi = require('joi')
+
+const schema = Joi.object({
+  personReference: Joi.string().required(),
+  firstName: Joi.string().required(),
+  lastName: Joi.string().required(),
+  dateOfBirth: Joi.date().iso().optional().allow('').allow(null),
+  address: Joi.object({
```
Comment (jbarnsley10) [inline app/schema/person/update.js:8]: Do we need max length checks on the fields or are you happy for the DB to catch that?

Comment (sfitz42) [reply app/schema/person/update.js:8]: Happy for the DB to catch that at the moment. Not a public facing API so probably not worth doing lots of work on errors.

File: app/repos/people.js (hunk 2 above)
Comment (jbarnsley10) [reply app/repos/people.js:112]: Ok, but do we have any documentation yet?

**Review by jbarnsley10 — APPROVED** [review]
LGTM

**Pull request merged**